controllers/storagecluster/topology.go

```diff
@@ -100,7 +100,7 @@ func setFailureDomain(sc *ocsv1.StorageCluster) {
 	// If sufficient zones are available then we select zone as the failure domain
 	topologyMap := sc.Status.NodeTopologies
 	for label, labelValues := range topologyMap.Labels {
-		if strings.Contains(label, "zone") {
+		if label == corev1.LabelZoneFailureDomainStable {
```

Comment:
label would not directly match like this, We need to first parse the label on the node to get the key part & then do the matching.
As the label on the node would in the form like
`
topology.kubernetes.io/zone=zone1 or topology.rook.io/rack=rack1 or kubernetes.io/hostname=node1
`
We need to parse the label first to get the key part of the label like 
`topology.kubernetes.io/zone or topology.rook.io/rack or kubernetes.io/hostname
`
And then we need to match them.

Reply:
![image](https://github.com/user-attachments/assets/66b927a5-f5ee-46db-8eca-41e90c7cacf3)
Based on my debug, the label and label values are separated.

Reply:
Oh, Sorry for the confusion, you are write in the node topology map the label contains just the key part.
found an example here 
```
nodeTopologies:
      labels:
        kubernetes.io/hostname:
        - k8sbm-1283800.ln.fw.gs.com
        - k8sbm-1283803.ln.fw.gs.com
        - k8sbm-1283804.ln.fw.gs.com
        - k8sbm-1283805.ln.fw.gs.com
        - k8sbm-1431842.k8s.gs.com
        - k8sbm-1431848.k8s.gs.com
        - k8sbm-1433161.k8s.gs.com
        - k8sbm-1433163.k8s.gs.com
        - k8sbm-1433165.k8s.gs.com
        - k8sbm-1433166.k8s.gs.com
        - k8sbm-1433178.k8s.gs.com
        - k8sbm-1433184.k8s.gs.com
        - k8sbm-1433185.k8s.gs.com
        - k8sbm-1433321.k8s.gs.com
        - k8sbm-1433322.k8s.gs.com
        - k8sbm-1433333.k8s.gs.com
        topology.rook.io/rack:
        - rack0
        - rack1
        - rack2
```